Microsoft.Azure.CosmosRepository/src/FullItem.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.CosmosRepository
+{
+    /// <summary>
+    /// A base helper class that implements IItemWithEtag
+    /// <remarks>
+    /// This base class will implement all supported interfaces giving the complete information available from this SDK.
+    /// </remarks>
+    /// </summary>
+    /// <example>
+    /// Here is an example subclass item, which adds several properties:
+    /// <code language="c#">
+    /// <![CDATA[
+    /// public class SubItem : FullItem
+    /// {
+    ///     public DateTimeOffset Date { get; set; }
+    ///     public string Name { get; set; }
+    ///     public IEnumerable<Child> Children { get; set; }
+    ///     public IEnumerable<string> Tags { get; set; }
+    /// }
+    ///
+    /// public class Child
+    /// {
+    ///     public string Name { get; set; }
+    ///     public DateTime BirthDate { get; set; }
+    /// }
+    /// ]]>
+    /// </code>
+    /// </example>
+    public abstract class FullItem : Item, IItemWithEtag
```

Comment:
Why is this `abstract` and what's the difference between this and the `EtagItem`? Why not inherit from `EtagItem` here, and add the two date properties?

Reply:
Does it depend on what the plan is I suppose do we want the time stamp changes in this PR? I thought we could do that as a separate piece of work, then we can have the `DatedItem` as an option for time stamps without the `ETagItem` then the `FullItem` could include all of `Item`, `DatedItem` & `ETagItem` 😃  it a lot of base types I know but it offers the most flexibility from a user point of view, what do you both think? Makes sense that this would not strictly *have* to be `abstract` as well I suppose.

Reply:
Sounds good to me, thanks for the clarification.

Reply:
As for the `abstract` we had a chat and did not see a use case for wanted an `IRepository<FullItem>` just on it's own with nothing else so we chose to leave it as is, will hang fire with the release just confirm :)